fastiron/src/constants.rs

```diff
@@ -45,105 +45,116 @@ pub type Tuple4 = (usize, usize, usize, usize);
+    + SubAssign
+    + MulAssign
+    + DivAssign
+    + Sum
+    // display
     + Debug
     + Display
     + LowerExp
-    + FromStr
-    + From<f32>
-    + Sum
-    + for<'a> Sum<&'a Self>
+    // parallel-safe
     + Send
     + Sync
 {
-}
-/// Custom super-trait for floating point number
-pub trait CustomFloat: Float + FromPrimitive + OpsFloat + UtilsFloat {
-    // floating ref
-
     /// Threshold upper-value for decimal number.
-    fn huge_float<T: CustomFloat>() -> T;
-    /// Threshold low-ish-value for decimal number.
-    fn small_float<T: CustomFloat>() -> T;
-    /// Threshold lower-value for decimal number.
-    fn tiny_float<T: CustomFloat>() -> T;
-
-    // physical constants
-
-    /// Neutron mass energy equivalent (MeV)
-    fn neutron_mass_energy<T: CustomFloat>() -> T;
-    /// [Pick your definition][3]
-    ///
-    /// [3]: https://en.wikipedia.org/wiki/Pi
-    fn pi<T: CustomFloat>() -> T;
-    /// Speed of light (cm/s)
-    fn light_speed<T: CustomFloat>() -> T;
-}
-
-impl OpsFloat for f32 {}
-impl UtilsFloat for f32 {}
-impl CustomFloat for f32 {
-    /// Threshold value for decimal number when using [f32]. May need adjustment.
     fn huge_float<T: CustomFloat>() -> T {
-        FromPrimitive::from_f32(10e35_f32).unwrap()
+        T::from(T::HUGE_FLOAT).unwrap()
     }
-
-    /// Threshold value for decimal number when using [f32]. May need adjustment.
+    /// Threshold low-ish-value for decimal number.
     fn small_float<T: CustomFloat>() -> T {
-        FromPrimitive::from_f32(1e-10_f32).unwrap()
+        T::from(T::SMALL_FLOAT).unwrap()
     }
-
-    /// Threshold value for decimal number when using [f32]. May need adjustment.
+    /// Threshold lower-value for decimal number.
     fn tiny_float<T: CustomFloat>() -> T {
-        FromPrimitive::from_f32(1e-13_f32).unwrap()
+        T::from(T::TINY_FLOAT).unwrap()
     }
 
     fn neutron_mass_energy<T: CustomFloat>() -> T {
-        FromPrimitive::from_f32(9.39565e+2).unwrap()
+        T::from(9.39565e+2).unwrap()
     }
 
     fn pi<T: CustomFloat>() -> T {
-        FromPrimitive::from_f32(std::f32::consts::PI).unwrap()
+        T::from(T::PI).unwrap()
     }
 
     fn light_speed<T: CustomFloat>() -> T {
-        FromPrimitive::from_f32(2.99792e+10).unwrap()
+        T::from(2.99792e+10).unwrap()
     }
 }
 
-impl OpsFloat for f64 {}
-impl UtilsFloat for f64 {}
-impl CustomFloat for f64 {
-    /// Threshold value for decimal number when using [f64].
-    fn huge_float<T: CustomFloat>() -> T {
-        FromPrimitive::from_f64(10e75_f64).unwrap()
-    }
+impl CustomReferenceFloat for f32 {
+    /// Threshold value for decimal number when using [f32]. May need adjustment.
+    const HUGE_FLOAT: Self = 10e35_f32;
+
+    /// Threshold value for decimal number when using [f32]. May need adjustment.
+    const SMALL_FLOAT: Self = 1e-10_f32;
 
+    /// Threshold value for decimal number when using [f32]. May need adjustment.
+    const TINY_FLOAT: Self = 1e-13_f32;
```

Comment:
Is it not defined in the standard library?

By the way, it is not really a decimal number as it is likely encoded in base 2.

Reply:
I removed the "decimal number" wording since it wasn't necessary in the first place. 

The value defined in the standard library are different from these ones, which are the same as the ones used by Quicksilver. using those from `std` results in different outputs when running simulations.